### DISCUSSION TITLE
Docker-Desktop: Volume mount not working

### DISCUSSION THREAD

**Discussion by pogopaule** [nix-community/NixOS-WSL#148] [Q&A] [answered]
I run release https://github.com/nix-community/NixOS-WSL/releases/tag/22.05-5c211b47 which at the time of writing is the latest and already includes [this bugfix](https://github.com/nix-community/NixOS-WSL/pull/117).

If I run `docker run --interactive --tty --rm --publish 34921:8080 --volume ${PWD}:/usr/local/structurizr structurizr/lite` the folder `/usr/local/structurizr` does not contain any of the files in `pwd`.

If I run the same command in WSL with Ubuntu, everything works fine. The folder is mounted correctly.

This is my configuration.nix

```
{ lib, pkgs, config, modulesPath, ... }:

with lib;
let
  nixos-wsl = import ./nixos-wsl;
in
{
  imports = [
    "${modulesPath}/profiles/minimal.nix"

    nixos-wsl.nixosModules.wsl
  ];

  wsl = {
    enable = true;
    automountPath = "/mnt";
    defaultUser = "pogopaule";
    startMenuLaunchers = true;

    # Enable native Docker support
    # docker-native.enable = true;

    # Enable integration with Docker Desktop (needs to be installed)
    docker-desktop.enable = true;

  };

  # Enable nix flakes
  nix.package = pkgs.nixFlakes;
  nix.extraOptions = ''
    experimental-features = nix-command flakes
  '';

  system.stateVersion = "22.05";
}
```

What am I doing wrong?

**Comment by nzbr** [comment, marked as answer]
Your config and the docker command both look fine to me. Can you try updating to the current state on `main`?

**Reply by pogopaule** [reply]
> Can you try updating to the current state on main?

How do I do so? I just know how to install releases via https://github.com/nix-community/NixOS-WSL#quick-start

**Reply by nzbr** [reply]
Just replace everything in `/etc/nixos/nixos-wsl` with a fresh copy of this repo and run `nixos-rebuild switch`. After that you should restart the WSL VM by running `wsl.exe --shutdown` on windows

**Reply by pogopaule** [reply]
Ok, that solves the problem. Do you have an idea why? I definitely installed the latetest release. I checked it using the sha256.

**Reply by nzbr** [reply]
The latest release is a few commits behind main and one of them is a full rewrite of the syschdemd script that is responsible for changing into the namespace where systemd runs. So the bug seems to have been fixed by the rewrite and fixing the old version wouldn't make any sense
